awscli/customizations/s3/filegenerator.py

```diff
@@ -130,11 +130,24 @@ def call(self, files):
         ``dir_op`` and ``use_src_name`` flags affect which files are used and
         ensure the proper destination paths and compare keys are formed.
         """
-        function_table = {'s3': self.list_objects, 'local': self.list_files}
         source = files['src']['path']
         src_type = files['src']['type']
         dest_type = files['dest']['type']
-        file_list = function_table[src_type](source, files['dir_op'])
+        if src_type == 'local':
+            direction_context = src_type + dest_type
+            # If the file generator has no associated operation, it
+            # is not the driving force behind an operation. That means
+            # it is being used as the other generator (the one being used
+            # for listing of files in the destination) in a sync command.
+            # More specifically this if statement will only be hit if it
+            # is the file generator listing files for a downloading sync.
+            if self.operation_name == '':
```

Comment:
I feel that this was sort of hacky when I wrote it as I am relying on the fact in the code when I [instantiate](https://github.com/aws/aws-cli/blob/develop/awscli/customizations/s3/subcommands.py#L758) the FileGenerator for the reference files in the destination, the ``operation_name`` is set to ``''``. I want some feedback on this part.

The other option I was thinking of  was to introduce a new argument to the constructor that specifies the transfer direction. The reason I did not do it was because it would bloat the constructor and may require a bunch of updates to the tests. Note I am not too concerned about coupling to the command being run here because the FileGenerator already requires an operation name.